src/org/gk/qualityCheck/QACheckUtilities.java

```diff
@@ -26,4 +33,297 @@ public static GKInstance getLatestCuratorIEFromInstance(GKInstance instance) thr
+                // QA checks calling these methods are concerned with PhysicalEntities that have a species attribute.
+                if (hasSpeciesAttribute(attributePE)) {
+                    inputOutputPEs.add(attributePE);
+                    inputOutputPEs.addAll(findAllPhysicalEntities(attributePE));
+                }
+            }
+        }
+        return inputOutputPEs;
+    }
+
+    /**
+     * Finds all PhysicalEntities that exist in a ReactionlikeEvent's catalystActivity. It checks the catalyst's
+     * activeUnit and physicalEntity attributes.
+     * @param reaction GKInstance -- ReactionlikeEvent that is being searched for PhysicalEntities.
+     * @return Set<GKInstance> -- All distinct PhysicalEntities in ReactionlikeEvent's catalystActivity.
+     * @throws Exception -- Thrown by MySQLAdaptor
+     */
+    public static Set<GKInstance> findAllCatalystPEs(GKInstance reaction) throws Exception {
+        Set<GKInstance> catalystPEs = new HashSet<>();
+        List<GKInstance> catalysts = reaction.getAttributeValuesList(ReactomeJavaConstants.catalystActivity);
+        for (GKInstance catalyst : catalysts) {
+            for (String attribute : Arrays.asList(ReactomeJavaConstants.activeUnit, ReactomeJavaConstants.physicalEntity)) {
+                for (GKInstance attributePE : (Collection<GKInstance>) catalyst.getAttributeValuesList(attribute)) {
+                    // QA checks calling these methods are concerned with PhysicalEntities that have a species attribute.
+                    if (hasSpeciesAttribute(attributePE)) {
+                        catalystPEs.add(attributePE);
+                        catalystPEs.addAll(findAllPhysicalEntities(attributePE));
+                    }
+                }
+            }
+        }
+        return catalystPEs;
+    }
+
+    /**
+     * Finds all PhysicalEntities that exist in a ReactionlikeEvent's regulatedBy. It checks the regulation's
+     * activeUnit and regulator attributes.
+     * @param reaction GKInstance -- ReactionlikeEvent that is being searched for PhysicalEntities.
+     * @return Set<GKInstance> -- All distinct PhysicalEntities in ReactionlikeEvent's regulatedBy.
+     * @throws Exception -- Thrown by MySQLAdaptor
+     */
+    public static Set<GKInstance> findAllRegulationPEs(GKInstance reaction) throws Exception {
+        Set<GKInstance> regulationPEs = new HashSet<>();
+        List<GKInstance> regulations = reaction.getAttributeValuesList(ReactomeJavaConstants.regulatedBy);
+        for (GKInstance regulation : regulations) {
+            for (String attribute : Arrays.asList(ReactomeJavaConstants.activeUnit, ReactomeJavaConstants.regulator)) {
+                for (GKInstance attributePE : (Collection<GKInstance>) regulation.getAttributeValuesList(attribute)) {
+                    // QA checks calling these methods are concerned with PhysicalEntities that have a species attribute.
```

Comment:
Could use a helper method similar to the triple for loop in catalyst physical entities

Reply:
Same thing as I commented above for the similar catalyst situation